app/src/main/java/space/taran/arknavigator/utils/FileUtils.kt

```diff
@@ -18,38 +20,48 @@ enum class Sorting {
     DEFAULT, NAME, SIZE, LAST_MODIFIED, TYPE
 }
 
-enum class FileActionType{
+enum class FileActionType {
     EDIT_AS_OPEN, EDIT_AND_OPEN, OPEN_ONLY, OPEN_ONLY_DETACH_PROCESS
 }
 
-private val acceptedImageExt = listOf(".jpg", ".jpeg", ".png")
-private val acceptedEditOnlyExt = arrayListOf(".txt", ".doc", ".docx", ".odt", "ods")
+const val PDF_PREVIEW_FOLDER_NAME = "pdf_preview"
+
+private val acceptedImageExt = listOf("jpg", "jpeg", "png")
+private val acceptedVideoExt = listOf("mp4", "avi", "mov", "wmv", "flv")
+private val acceptedEditOnlyExt = arrayListOf("txt", "doc", "docx", "odt", "ods")
     .also { it.addAll(acceptedImageExt) }
-private val acceptedReadAndEditExt = listOf(".pdf", ".md")
-private val acceptedVideoExt = listOf(".mp4", ".wmv", ".avi", ".flv", ".mkv")
+private val acceptedReadAndEditExt = listOf("pdf", "md")
 
-fun provideIconImage(file: Path): Path? =
-    providePreview(file) //todo downscale to, say, 128x128
+fun isImage(filePath: Path?): Boolean {
+    val extension = extension(filePath)
+    return acceptedImageExt.contains(extension)
+}
 
-//might be a temporary file
-fun providePreview(file: Path): Path? =
-    if (isImage(file)) {
-        file
+fun isVideo(filePath: Path?): Boolean {
+    return if (filePath?.toFile()?.exists() == true && filePath.fileSize() > 0) {
```

Comment:
We must not call method `isVideo` if the file doesn't exist. We should handle this somewhere else, irrelevant on file type.

Reply:
Also, don't use `java.io.File`, use `java.nio.Path`. All methods like `exists` are possible with newer `java.nio` API.